flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -412,7 +412,7 @@ public void registerType(Class<?> type) {
 		if (from > to) {
 			throw new IllegalArgumentException("Start of sequence must not be greater than the end");
 		}
-		return fromCollection(new NumberSequenceIterator(from, to), BasicTypeInfo.LONG_TYPE_INFO, "Sequence Source");
+		return addSource(new StatefulSequenceSource(from, to), "Sequence Source").setParallelism(1);
 	}
```

Comment:
The current behaviour was that if the user tried setting at least two as parallelism then the stream graph generator throw an exception on execute, telling the user that this is in fact not a parallel source.
If we enable setting it the way you proposed than it is not much use the having a separate `generateSequence` and a `generateParallelSequence`. Personally I am not a big fan of the distinction, but if it is there let us do it right. Are the batch versions different?
This is not a release blocker though, a JIRA for 0.10 is an OK solution on my side.